pkg/azurefile/controllerserver.go

```diff
@@ -927,12 +963,26 @@ func (d *Driver) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshotRequ
 		mc.ObserveOperationWithResult(isOperationSucceeded, SnapshotID, req.SnapshotId)
 	}()
 
-	if _, err := shareURL.WithSnapshot(snapshot).Delete(ctx, azfile.DeleteSnapshotsOptionNone); err != nil {
-		if strings.Contains(err.Error(), "ShareSnapshotNotFound") {
+	var deleteErr error
+	// if len(req.GetSecrets()) > 0 {
+	// 	shareURL, err := d.getShareURL(ctx, req.SnapshotId, req.GetSecrets())
+	// 	if err != nil {
+	// 		// According to CSI Driver Sanity Tester, should succeed when an invalid snapshot id is used
+	// 		klog.V(4).Infof("failed to get share url with (%s): %v, returning with success", req.SnapshotId, err)
+	// 		return &csi.DeleteSnapshotResponse{}, nil
+	// 	}
+
+	// 	_, deleteErr = shareURL.WithSnapshot(snapshot).Delete(ctx, azfile.DeleteSnapshotsOptionNone)
+	// } else {
+	deleteErr = d.cloud.FileClient.WithSubscriptionID(subsID).DeleteFileShare(ctx, rgName, accountName, fileShareName, snapshot)
```

Comment:
looks like this is deleting all snapshots under one file share, original logic is only deleting one snapshot